ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/user-notifications/user-notifications.view.js

```diff
@@ -0,0 +1,116 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import React from 'react'
+import styled from '../../styles/styled-components'
+import withListenTo from '../../container/backbone-container'
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+var Backbone = require('backbone')
+var CustomElements = require('../../../js/CustomElements.js')
+var NotificationGroupView = require('../../../component/notification-group/notification-group.view.js')
+var user = require('../../../component/singletons/user-instance.js')
+var moment = require('moment')
+var userNotifications = require('../../../component/singletons/user-notifications.js')
+
+const Empty = styled.div`
+  transition: transform ${props => props.theme.coreTransitionTime} linear;
+  transform: scale(1);
+  text-align: center;
+  font-size: ${props => props.theme.largeFontSize}
+  padding: ${props => props.theme.mediumSpacing}
+`
+const Notifications = styled.div`
+  height: 100%;
+  width: 100%;
+  display: block;
+  overflow: auto;
+  padding: ${props => props.theme.mediumSpacing};
+`
+
+class UserNotifications extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = userNotifications
+    this.props.listenTo(
+      userNotifications,
+      'add remove update',
+      this.updateState.bind(this)
+    )
+  }
+  render() {
+    let notificationsList = []
+    for (let i = 0; i <= 8; i++) {
+      notificationsList.push(
+        <MarionetteRegionContainer
+          key={i.toString()}
+          view={this.listPreviousDays(i)}
+          viewOptions={this.replaceElement()}
+        />
+      )
+    }
+    return userNotifications.isEmpty() ? (
+      <Empty>
+        <div>No Notifications</div>
+      </Empty>
+    ) : (
+      <Notifications>{notificationsList}</Notifications>
+    )
+  }
+  updateState() {
+    this.setState(userNotifications)
+  }
+  listPreviousDays(numDays) {
+    if (numDays < 7) {
+      return new NotificationGroupView({
+        filter: function(model) {
+          return moment().diff(model.get('sentAt'), 'days') === numDays
+        },
+        date: this.informalName(numDays),
+      })
+    } else {
+      return new NotificationGroupView({
+        filter: function(model) {
+          return moment().diff(model.get('sentAt'), 'days') >= 7
+        },
+        date: 'Older',
+      })
+    }
+  }
+  informalName(daysAgo) {
+    switch (daysAgo) {
+      case 0:
+        return 'Today'
+        break
+      case 1:
+        return 'Yesterday'
+        break
+      default:
+        return moment()
+          .subtract(daysAgo, 'days')
+          .format('dddd')
+        break
+    }
+  }
+  replaceElement() {
+    return { replaceElement: true }
+  }
+  onDestroy() {
```

Comment:
❌ onDestroy is a Marionette lifecycle function - this won't do anything unless called.
`componentWillUnmount()` is what you need here.